mplc/splitter.py

```diff
@@ -37,29 +33,41 @@ def __str__(self):
     def split(self, partners_list, dataset):
         self.dataset = dataset
         self.partners_list = partners_list
-        if len(self.amounts_per_partner) != self.partners_count:
-            raise AttributeError(f"The amounts_per_partner list should have a size ({len(self.amounts_per_partner)}) "
-                                 f"equals to partners_count ({self.partners_count})")
 
-        logger.info("### Splitting data among partners:")
-        logger.info("Train data split:")
+        logger.info("Splitting data among partners: starting now.")
+        self._test_config_coherence()
+        logger.info("Coherence of config parameters: OK.")
+
+        logger.info("Train data split: starting now.")
         self._split_train()
 
         if self.val_set == 'local':
-            logger.info("Validation data split:")
+            logger.info("Validation data split: starting now.")
             self._split_val()
 
         if self.test_set == 'local':
-            logger.info("Test data split:")
+            logger.info("Test data split: starting now.")
             self._split_test()
 
         for partner in self.partners_list:
             logger.info(
-                f"   Partner #{partner.id}: "
-                f"{partner.final_nb_samples} samples "
-                f"with labels {partner.labels}"
+                f"Partner #{partner.id}: {partner.final_nb_samples} samples with labels {partner.labels}"
             )
 
+    @abstractmethod
```

Comment:
Technically an abstract method should never be called (even if it seems that python allows it), but must be re-defined by the subclasses. 

It is a way of saying, 'hey, you are my subclass, you should implement your own $abstractmethod$'
kind of a template.

I maybe misunderstood but I don't think that is your objective ? 

Typically all the other splitters could reuse the method implemented in the abstract class if this method wasn't abstract

Reply:
I am bit torn apart here - initially I wanted indeed to force all subclass to reimplement it, as these coherence tests are very important and could be quite different depending on the splitter. But it turned out 3 out of the current 4 use the same... I think I'll relax it, and we'll see in the future if it is best to re-make it an abstract method.